sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferStore.scala

```diff
@@ -304,6 +320,34 @@ abstract class RapidsBufferStore(
       }
     }
 
+    override def getDeviceMemoryBuffer: DeviceMemoryBuffer = {
+      (0 until MAX_UNSPILL_ATTEMPTS).foreach { _ =>
+        catalog.acquireBuffer(id, DEVICE) match {
+          case Some(buffer) =>
+            withResource(buffer) { buffer =>
+              return buffer.getDeviceMemoryBuffer
+            }
+          case _ =>
+            try {
+              val newBuffer = {
+                logDebug(s"Unspilling $this $id to $DEVICE")
+                RapidsBufferCatalog.getDeviceStorage.copyBuffer(
+                  this, getMemoryBuffer, Cuda.DEFAULT_STREAM)
+              }
+              if (newBuffer.addReference()) {
+                withResource(newBuffer) { newBuffer =>
+                  return newBuffer.getDeviceMemoryBuffer
+                }
+              }
+            } catch {
+              case e: IllegalStateException =>
+                logDebug("Error unspilling to device", e)
```

Comment:
I don't think we want to log the full exception stack here and call it an error when it is expected and will be retried.  If it does occur it will add a lot of spam to the debug log.  Maybe something along the line of `s"Lost device buffer registration race for buffer $id, retrying..."`